### ISSUE TITLE
Unable to run inference with exported SDXL model

### ISSUE THREAD

**Issue by Paulkm2006** [huggingface/optimum-neuron#221]
Hi there, 

I used the neuron exporter described in [this documentation](https://huggingface.co/docs/optimum-neuron/guides/export_model#exporting-stable-diffusion-xl-to-neuron) to export the model into neuron format. 

The command I used is
`optimum-cli export neuron --model stabilityai/stable-diffusion-xl-base-1.0 --task stable-diffusion-xl --batch_size 1 --height 1024 --width 1024 --auto_cast matmul --auto_cast_type bf16 sdxl_neuron_pipe/`

Since I don't have enough quota to run it on a inf2 instance with sufficient memory (took about 60gb max), I performed this step on another machine without neuron. In general, the process was successful except the error in the end, when the exporter tried to load the model into a neuron device and test it.

[This](https://huggingface.co/paulkm/sdxl_neuron_pipe) is the model I get.

But when I tried to load it into a inf2 instance, I got the following exception:
```
---------------------------------------------------------------------------
IndexError                                Traceback (most recent call last)
Cell In[2], line 3
      1 compiler_args = {"auto_cast": "matmul", "auto_cast_type": "bf16"}
      2 input_shapes = {"batch_size": 1, "height": 1024, "width": 1024}
----> 3 pipe = NeuronStableDiffusionXLPipeline.from_pretrained("sdxl_neuron_pipe", device_ids=[0,1], **compiler_args, **input_shapes)

File ~/.local/lib/python3.10/site-packages/optimum/modeling_base.py:372, in OptimizedModel.from_pretrained(cls, model_id, export, force_download, use_auth_token, cache_dir, subfolder, config, local_files_only, trust_remote_code, revision, **kwargs)
    369     trust_remote_code = False
    371 from_pretrained_method = cls._from_transformers if export else cls._from_pretrained
--> 372 return from_pretrained_method(
    373     model_id=model_id,
    374     config=config,
    375     revision=revision,
    376     cache_dir=cache_dir,
    377     force_download=force_download,
    378     use_auth_token=use_auth_token,
    379     subfolder=subfolder,
    380     local_files_only=local_files_only,
    381     trust_remote_code=trust_remote_code,
    382     **kwargs,
    383 )

File ~/.local/lib/python3.10/site-packages/optimum/neuron/modeling_diffusion.py:418, in NeuronStableDiffusionPipelineBase._from_pretrained(cls, model_id, config, use_auth_token, revision, cache_dir, text_encoder_file_name, text_encoder_2_file_name, unet_file_name, vae_encoder_file_name, vae_decoder_file_name, local_files_only, model_save_dir, device_ids, **kwargs)
    415         configs[name] = model_config
    416         neuron_configs[name] = cls._neuron_config_init(model_config)
--> 418 text_encoder, unet, vae_decoder, vae_encoder, text_encoder_2 = cls.load_model(
    419     text_encoder_path=model_and_config_save_paths["text_encoder"][0],
    420     unet_path=model_and_config_save_paths["unet"][0],
    421     vae_decoder_path=model_and_config_save_paths["vae_decoder"][0],
    422     vae_encoder_path=model_and_config_save_paths["vae_encoder"][0],
    423     text_encoder_2_path=model_and_config_save_paths["text_encoder_2"][0],
    424     device_ids=device_ids,
    425     dynamic_batch_size=neuron_configs[DIFFUSION_MODEL_UNET_NAME].dynamic_batch_size,
    426 )
    428 if model_save_dir is None:
    429     model_save_dir = new_model_save_dir

File ~/.local/lib/python3.10/site-packages/optimum/neuron/modeling_diffusion.py:246, in NeuronStableDiffusionPipelineBase.load_model(text_encoder_path, unet_path, vae_decoder_path, vae_encoder_path, text_encoder_2_path, device_ids, dynamic_batch_size)
    243 text_encoder = NeuronBaseModel.load_model(text_encoder_path)
    244 if len(device_ids) > 1:
    245     unet = torch_neuronx.DataParallel(
--> 246         torch.jit.load(unet_path),
    247         device_ids,
    248         set_dynamic_batching=dynamic_batch_size,
    249     )
    250 else:
    251     unet = NeuronBaseModel.load_model(unet_path)

File ~/.local/lib/python3.10/site-packages/torch/jit/_serialization.py:162, in load(f, map_location, _extra_files)
    160 cu = torch._C.CompilationUnit()
    161 if isinstance(f, str) or isinstance(f, pathlib.Path):
--> 162     cpp_module = torch._C.import_ir_module(cu, str(f), map_location, _extra_files)
    163 else:
    164     cpp_module = torch._C.import_ir_module_from_buffer(
    165         cu, f.read(), map_location, _extra_files
    166     )

IndexError: vector::_M_range_check: __n (which is 188) >= this->size() (which is 5)
```

The weird part is that the model I got following [this notebook](https://github.com/aws-neuron/aws-neuron-samples/blob/master/torch-neuronx/inference/hf_pretrained_sdxl_1024_inference.ipynb) worked well, and when I tried to replace the newly-generated model.neuron with the one I got from the notebook, the pipeline loaded successfully, but unfortunately unable to perform any inference.

**Comment by JingyaHuang** [comment]
Hey @Paulkm2006, thanks for reporting. 

Are you using the main branch of `optimum-neuron` for what you describe above? The error likely comes from a compiler args not supported yet by the unet of SDXL, could you check if the flag  `--enable-fast-loading-neuron-binaries` is on? Thx.

**Comment by JingyaHuang** [comment]
Meantime will try to reproduce your issue, can you send me what instance type, os version and the environment setup?

You can use the following commands:
```
pip3 list | grep -e neuron -e xla -e torch -e optimum -e diffusers
apt list --installed | grep aws-neuron
```

**Comment by JingyaHuang** [comment]
@Paulkm2006 Thanks for the details, it seems that your compilation env is not using the latest main branch. 

The main branch should be compatible with optimum 1.13.1 through this PR #217, also the issue caused by `--enable-fast-loading-neuron-binaries` flag shall has been fixed since I merge the PR #212. Could you uninstall `optimum-neuron`, reinstall it with the following command and try again? Thx

```
pip install https://github.com/huggingface/optimum-neuron.git
```
